promql/engine.go

```diff
@@ -385,16 +385,21 @@ func (ng *Engine) NewRangeQuery(q storage.Queryable, qs string, start, end time.
 	return qry, nil
 }
 
-func (ng *Engine) newQuery(q storage.Queryable, expr parser.Expr, start, end time.Time, interval time.Duration) (*query, error) {
+func (ng *Engine) newQuery(q storage.Queryable, expr parser.Expr, start, end time.Time, interval, lookbackDelta time.Duration) (*query, error) {
 	if err := ng.validateOpts(expr); err != nil {
 		return nil, err
 	}
 
+	if lookbackDelta == 0 {
```

Comment:
```suggestion
	if lookbackDelta <= 0 {
```

let's safeguard this

Reply:
https://github.com/prometheus/prometheus/pull/9946/commits/e79ef5b00523ca6226bf92fdb0869c7ddda570ec